autobuild/workflow.md

```diff
@@ -0,0 +1,195 @@
+* Click the green *Run workflow* button.
+
+If successful, the workflow takes around 30 minutes to build and push a new image to Docker hub. The image will have a -staging tag at this point signifying it has not been tested and released for public use.
+
+## Build exrproxy-env deployment package
+**Audience:**
+Not applicable to 3rd party developers who are simply looking to list a coin.
+
+* These steps are based closely on https://docs.blocknet.co/service-nodes/setup/#prepare-to-deploy-service-node.
+* Clone the https://github.com/blocknetdx/exrproxy-env repo to your local machine. You need to create a deploy script for the servicenode which will oversee your test trades, and a deploy script for the two trading nodes.
+* cd exrproxy-env/autobuild
+* cp examples/alldaemons.yaml custom.yaml
+* Edit custom.yaml to create a (testnet) servicenode which includes the coin to be tested and any counter-party coin you want to use.
+* Run app.py specifying the path for your branch pushed at step 8 above, eg:
+```
+python3 app.py --branchpath https://raw.githubusercontent.com/walkjivefly/blockchain-configuration-files/bump-thing-v4.2.2
+```
+* Copy the newly created dockercompose-custom.yaml to your servicenode machine.
+* Edit custom.yaml to create a (testnet) trading node.
+* Run app.py again, as above, to create the dockercompose-custom.yaml to be used on the trading machines.
+* Copy the trading dockercompose-custom.yaml to the trading machines. 
+
+## Testing (CLI)
+**Audience:**
+Not directly applicable to 3rd party developers, although they could choose to follow a similar process if they don't want to jump through the hoops required (at this time) to test with BlockDX.
+
+* Bring up the Docker containers on all 3 machines and let them sync if necessary.
+* Fund, encrypt and unlock the wallets as required.
+* Restart the servicenode as a servicenode.
+* On one trading machine attempt an XBridge trade between your new coin and any counter-party using dxmakeorder.
+* Check the order is visible on the other trading machine and attempt to take it using dxtakeorder.
+* Confirm several orders complete successfully, check that interrupted trades are correctly rolled back and all funds end up where they should. Record the trade ids for inclusion in the pull request. 
+
+## Testing (BlockDX)
+Instructions coming soon (TM).
+
+Comment: this is where https://github.com/blocknetdx/block-dx/issues/339 would be really useful right now!
+
+## Upon successful testing
+**Audience**: everyone
+
+* Open a pull request in https://github.com/blocknetdx/blockchain-configuration-files
+* Once approved the new configuration information will be merged into the master branch. An automated workflow will copy it out to AWS from where it will be accessible to any BlockDX user or other interested parties.
+* QA engineer will run the RELEASE IMAGE action in https://github.com/blocknetdx/dockerimages to build a production image without the -staging tag.
```

Comment:
Here again, I think we probably want to give your Discord ID & mine as "QA Engineers" who can run the RELEASE IMAGE workflow. And/Or direct them to the #block-dx-listing channel in Blocknet Discord and ask them to post a request there.

Reply:
hmm, if they've got this far and opened a PR then they've made contact. I imagine we'd want to build and test/confirm for ourselves the config is good, the Docker image works and it's safe to merge the PR.

Reply:
Yeah, good point that they'll already have a way of contacting us if they've reached this far. No need to provide contact info again here.

>  I imagine we'd want to build and test/confirm for ourselves the config is good, the Docker image works and it's safe to merge the PR.

That's a very good question. Do we really want to take responsibility for verifying all coins traded on XBridge are properly tested & verified? Of course it will be nice if that's the case, but placing that responsibility on us makes the system more *centralized*. Maybe a little centralization isn't so bad in this case...?  I don't have a strong opinion atm. Need to ponder...